### ISSUE TITLE
Gun texture is 2x size in vk_nearest mode

### ISSUE THREAD

**Issue by lafoxxx92** [yquake2/yquake2#700]
build:
8.00pre

steps:
vid_renderer vk
vid_restart
vk_txturemode vk_nearest

actual:
looks like texture size is larger, it's resampled
![image](https://user-images.githubusercontent.com/34402711/114314636-1df98b80-9b04-11eb-9172-99538d110881.png)

expected:
looks same as when using gl1, gl3 or soft
![image](https://user-images.githubusercontent.com/34402711/114314654-2b167a80-9b04-11eb-8bc2-b808f666dd80.png)

**Comment by Spirrwell** [comment]
It looks like the Vulkan implementation always scales up to the nearest power of 2 texture in the Vk_Upload32 function.

It looks like the GL3 implementation simply doesn't bother. 

The GL1 implementation checks against gl_config.npottextures, which is set from checking for the GL_ARB_texture_non_power_of_two extension.

Should the Vulkan implementation just follow GL3 and not bother scaling?

**Comment by lafoxxx92** [comment]
Scaling should not be enabled by default, because "VK_NEAREST" (just like "GL_NEAREST") is **never** on by default, and people (like me) only enable this mode to make the game look pixelated (like in Software). Having this scaling enabled messes up the textures, and makes them look even worse than in Software, where everything is nice and sharp.

**Comment by Spirrwell** [comment]
As an extra note, removing the nearest power of 2 scaling to be parity with GL3 would also make vk_picmip useless, but GL3 doesn't have picmip either it looks like.

**Comment by lafoxxx92** [comment]
My opinion might be ignorant, but today's computers are very powerful compared to 1997 ones, and disabling mipmapping might not affect performance and/or take much VRAM in any way.

**Comment by Spirrwell** [comment]
I don't think it would break mipmapping altogether, and it may even be possible to support vk_picmip still despite it being weird, but I'm trying to essentially ask what Yamagi wants.
